msgpack_pack.c

```diff
@@ -10,242 +10,172 @@
 #include "msgpack_errors.h"
 
 #include "msgpack/pack_define.h"
-#define msgpack_pack_user smart_str*
+#define msgpack_pack_user smart_string*
 #define msgpack_pack_inline_func(name) \
     static inline void msgpack_pack ## name
 #define msgpack_pack_inline_func_cint(name) \
     static inline void msgpack_pack ## name
 #define msgpack_pack_append_buffer(user, buf, len) \
-    smart_str_appendl(user, (const void*)buf, len)
+    smart_string_appendl(user, (const void*)buf, len)
 #include "msgpack/pack_template.h"
+#define Z_REF_AWARE_P(_obj) \
```

Comment:
This macro feels pointless, but I couldn't find a way to implement this functionality without a lot of extra code